libraries/Microsoft.Bot.StreamingExtensions/RequestEx.cs

```diff
@@ -0,0 +1,139 @@
+
+        /// <summary>
+        /// Creates a <see cref="Request"/> to get resources hosted on a remote server.
+        /// </summary>
+        /// <param name="path">Optional path where the resource can be found on the remote server.</param>
+        /// <param name="body">Optional body to send to the remote server.</param>
+        /// <returns>A <see cref="Request"/> with appropriate status code and body.</returns>
+        public static Request CreateGet(string path = null, HttpContent body = null)
+        {
+            return CreateRequest(GET, path, body);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="Request"/> to post data to a remote server.
+        /// </summary>
+        /// <param name="path">Optional path where the resource can be found on the remote server.</param>
+        /// <param name="body">Optional body to send to the remote server.</param>
+        /// <returns>A <see cref="Request"/> with appropriate status code and body.</returns>
+        public static Request CreatePost(string path = null, HttpContent body = null)
+        {
+            return CreateRequest(POST, path, body);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="Request"/> to put updated data on a remote server.
+        /// </summary>
+        /// <param name="path">Optional path where the resource can be found on the remote server.</param>
+        /// <param name="body">Optional body to send to the remote server.</param>
+        /// <returns>A <see cref="Request"/> with appropriate status code and body.</returns>
+        public static Request CreatePut(string path = null, HttpContent body = null)
+        {
+            return CreateRequest(PUT, path, body);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="Request"/> to delete data hosted on a remote server.
+        /// </summary>
+        /// <param name="path">Optional path where the resource can be found on the remote server.</param>
+        /// <param name="body">Optional body to send to the remote server.</param>
+        /// <returns>A <see cref="Request"/> with appropriate status code and body.</returns>
+        public static Request CreateDelete(string path = null, HttpContent body = null)
+        {
+            return CreateRequest(DELETE, path, body);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="Request"/> with the passed in method, path, and body.
+        /// </summary>
+        /// <param name="method">The HTTP verb to use for this request.</param>
+        /// <param name="path">Optional path where the resource can be found on the remote server.</param>
+        /// <param name="body">Optional body to send to the remote server.</param>
+        /// <returns>On success returns a <see cref="Request"/> with appropriate status code and body, otherwise returns null.</returns>
+        public static Request CreateRequest(string method, string path = null, HttpContent body = null)
+        {
```

Comment:
Make this private.

Reply:
But this is also callable by consumers in case they want to use these classes with other verbs